geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSetCommandsFunctionExecutor.java

```diff
@@ -164,6 +174,17 @@ public long zrevrank(RedisKey key, byte[] member) {
   @Override
   public long zunionstore(RedisKey destinationKey, List<ZKeyWeight> keyWeights,
       ZAggregator aggregator) {
+    List<RedisKey> keysToLock = lockKeys(destinationKey, keyWeights);
+
+    return stripedExecute(destinationKey, keysToLock,
+        () -> new RedisSortedSet(Collections.emptyList(), new double[] {}).zunionstore(
```

Comment:
If you add that new RedisSortedSet constructor that just takes a size then you should use it here with a size of 0.